crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -92,6 +95,23 @@ pub(super) async fn configure_dsn(
         piece_cache.clone(),
     );
 
+    let last_archived_segment_index = Arc::new(AtomicU64::default());
+    tokio::spawn({
+        let mut archived_segments_notifications =
+            node_client
+                .subscribe_archived_segments()
+                .await
+                .map_err(|err| anyhow::anyhow!(err.to_string()))?;
```

Comment:
I think you can use `.context("Failed to subscribe to archived segments")` instead.

Reply:
Basically you receive some abstract error (thing which implements `std::error::Error`), it can have its own `source()` (which is essentially an error within an error -- more info about it in [docs](https://doc.rust-lang.org/std/error/trait.Error.html#method.source)) and most of the time you actually remove that context if you do `err.to_string()`.

[`.context()` method](https://docs.rs/anyhow/1.0.69/anyhow/trait.Context.html#tymethod.context) would wrap the error with some additional error message, while leaving the original error there